### PR TITLE
feat: manual/cheapest research mode toggle

### LINKED ISSUE #95: Feature: Research mode toggle
Implement a manual research mode toggle to switch off the cheapest automatic research mode unlocked with Cube Upgrade 9:

image

### PR DESCRIPTION
**Pull request by nmussy**
Fixes #95

### CODE REVIEW

File: src/Synergism.ts [hunk 1]
```diff
@@ -1281,6 +1282,12 @@ const loadSynergy = () => {
         if (!player.autoResearchToggle) {
             document.getElementById("toggleautoresearch").textContent = "Automatic: OFF"
         }
+        if (player.autoResearchMode === 'cheapest') {
+            document.getElementById("toggleautoresearchmode").textContent = "Automatic mode: Cheapest"
+        }
+        if (player.autoResearchMode === 'manual') {
```
Comment (KhafraDev) [inline src/Synergism.ts:1288]:
```suggestion
        } else if (player.autoResearchMode === 'manual') {
```

File: src/UpdateHTML.ts [hunk 2]
```diff
@@ -306,6 +306,10 @@ export const revealStuff = () => {
         document.getElementById("toggleautoresearch").style.display = "block" :
         document.getElementById("toggleautoresearch").style.display = "none";
 
+    player.shopUpgrades.obtainiumAuto > 0 && player.cubeUpgrades[9] > 0 ? //Auto Research Shop Purchase Mode
+        document.getElementById("toggleautoresearchmode").style.display = "block" :
+        document.getElementById("toggleautoresearchmode").style.display = "none";
```
Comment (KhafraDev) [inline src/UpdateHTML.ts:311]:
```suggestion
    document.getElementById("toggleautoresearchmode").style.display = player.shopUpgrades.obtainiumAuto > 0 && player.cubeUpgrades[9] > 0 ? //Auto Research Shop Purchase Mode
        ? 'block'
        : 'none';
```

File: src/types/Synergism.d.ts [hunk 3]
```diff
@@ -347,6 +348,7 @@ export interface Player {
     autoFortifyToggle: boolean,
     autoEnhanceToggle: boolean,
     autoResearchToggle: boolean,
+    autoResearchMode: AutoResearchMode,
```
Comment (KhafraDev) [inline src/types/Synergism.d.ts:351]: in CheckVariables.ts please add a default value, ie:
```ts
if (data.autoResearchMode === undefined) {
    player.autoResearchMode = '...'
}
```

**Review by KhafraDev — DISMISSED** [review]

File: src/Synergism.ts (hunk 1 above)
Comment (nmussy) [reply src/Synergism.ts:1288]: I agree it's cleaner, but I've followed the style of every other if surrounding it, it's probably best to leave it as is for the sake of consistency

File: src/UpdateHTML.ts (hunk 2 above)
Comment (nmussy) [reply src/UpdateHTML.ts:311]: Again, consistency

File: src/types/Synergism.d.ts (hunk 3 above)
Comment (nmussy) [reply src/types/Synergism.d.ts:351]: I've bypassed the need for an explicit default there: 

https://github.com/Pseudo-Corp/SynergismOfficial/blob/1403ebdcb7e69867939bc3af59273ca3f2fa3e85/src/Toggles.ts#L405-L411

I can add an explicit one in `CheckVariables.ts` if you'd like

Comment (nmussy) [reply src/types/Synergism.d.ts:351]: Fixed by 7050fb2

File: src/Synergism.ts (hunk 1 above)
Comment (KhafraDev) [reply src/Synergism.ts:1289]: the code is rather messy and continuing bad styling isn't a good idea

File: src/UpdateHTML.ts (hunk 2 above)
Comment (KhafraDev) [reply src/UpdateHTML.ts:311]: consistency isn't always the best (I changed a few of those blocks above, so there isn't really any consistency whatsoever anyways)

**Review by KhafraDev — DISMISSED** [review]

**Comment by nmussy** [comment]
I don't think the CI action ran after it initially failed (https://github.com/Pseudo-Corp/SynergismOfficial/runs/3107257619), they're all pending "Action required": https://github.com/Pseudo-Corp/SynergismOfficial/actions

image

Not sure why, it should run on every push? https://github.com/Pseudo-Corp/SynergismOfficial/actions/runs/1046329615/workflow#L4

**Comment by KhafraDev** [comment]
> 
> 
> I don't think the CI action ran after it initially failed (https://github.com/Pseudo-Corp/SynergismOfficial/runs/3107257619), they're all pending "Action required": https://github.com/Pseudo-Corp/SynergismOfficial/actions
> 
> image
> 
> Not sure why, it should run on every push? https://github.com/Pseudo-Corp/SynergismOfficial/actions/runs/1046329615/workflow#L4

"First-time contributors need a maintainer to approve running workflows."

**Comment by KhafraDev** [comment]
it's just a syntax error, i'll re-review once it's fixed

File: src/types/Synergism.d.ts (hunk 3 above)
Comment (KhafraDev) [inline src/types/Synergism.d.ts:351]:
```suggestion
    autoResearchMode: 'cheapest' | 'manual'
```

File: src/types/Synergism.d.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { HepteractCraft } from '../Hepteracts';
 import { Category, ResetHistoryEntryUnion } from '../History';
 import { IPlatBaseCost } from '../Platonic';
 import type { QuarkHandler } from '../Quark';
+import { AutoResearchMode } from '../Research';
```
Comment (KhafraDev) [inline src/types/Synergism.d.ts:7]:
```suggestion

```

File: src/Research.ts [hunk 5]
```diff
@@ -4,6 +4,8 @@ import { revealStuff } from './UpdateHTML';
 import { Globals as G } from './Variables';
 import { updateClassList } from './Utility';
 
+export type AutoResearchMode = 'manual' | 'cheapest';
```
Comment (KhafraDev) [inline src/Research.ts:7]:
```suggestion

```

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: src/Research.ts (hunk 5 above)
Comment (nmussy) [reply src/Research.ts]: Yeah it was a little overkill, fixed 👍

**Review by KhafraDev — APPROVED** [review]

**Pull request merged**